src/app/debug/filter-side-drawer/filter.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import { debounceTime, Observable, Subject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FilterService {
+  private showFilterSubject: Subject<boolean> = new Subject();
+  private metadataNamesSubject: Subject<string[]> = new Subject();
+  private filterContextSubject: Subject<Record<string, string>> = new Subject();
+  private currentRecordsSubject: Subject<Map<string, Array<string>>> = new Subject();
+  private filters: Record<string, string> = {};
```

Comment:
Is there a reason that it's initialized with an empty object here?

Reply:
since I've changed the records all into Map types, this was no longer necessary.